File: python/matrix/matrix.py
class Matrix:
    def __init__(self, m):
        self.m = [[int(n) for n in r.split()] for r in m.split('\n')]

    def row(self, i):
        return self.m[i-1]

    def column(self, i):
        return [r[i-1] for r in self.m]

File: java/matrix/src/main/java/Matrix.java
class Matrix {
    int[][] rows, cols;

    Matrix(String matrix) {
        String[] lines = matrix.split("\n");

        int height = lines.length;
        int width = lines[0].split(" ").length;

        rows = new int[height][width];
        cols = new int[width][height];

        for (int x = 0; x < lines.length; x++) {
            String[] numbers = lines[x].split(" ");
            for (int y = 0; y < numbers.length; y++) {
                rows[x][y] = cols[y][x] = Integer.parseInt(numbers[y]);
            }
        }
    }

    int[] getRow(int index) {
        return rows[index];
    }

    int[] getColumn(int index) {
        return cols[index];
    }
}
